notifier.js

```diff
@@ -0,0 +1,33 @@
+const notifier = require('node-notifier');
+
+const notify = (msg) => {
+    notifier.notify(
+        {
+          title: 'Notification',
+          message: msg,
+          //icon: path.join(__dirname, 'coulson.jpg'), // Absolute path (doesn't work on balloons)
+          sound: true, // Only Notification Center or Windows Toasters
+          wait: true // Wait with callback, until user action is taken against notification, does not apply to Windows Toasters as they always wait
+        },
+        (err, response) => {
+          // Response is response from notification
+        }
+      );
```

Comment:
So this is a callback function, try to figure out how to Promise-ify it without an external library. It'll be a useful lesson for both of you and promises provide an easier asynchronous API to handle instead of callback functions. We'll learn working with Promises first by writing everything using them, including the upcoming HTTP client and server. Then we'll rewrite everything with async/await as it is just syntax sugar for Promises. (syntax sugar -> syntactical analogs that just improve readability)

Read about promises [here](https://javascript.info/promise-basics).

An example of a callback function being promisified is:
```
function someMain() {
	someModule.someFunctionThatProvidesACallBackArgument(options, (err, arg1, arg2) {
		return new Promise((resolve, reject) => {
			if(err) {
				reject(err);
			} else {
				// Do some processing
				resolve(someData);
			}
		});
	})
}
```

Reply:
A more advanced blog post: https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html

Give it a read once you've understood the previous one linked.